### DISCUSSION TITLE
Increase "Max Source" to 100+, /{folder}, and/or Whole Vault (for Zettelkasten purpose)

### DISCUSSION THREAD

**Discussion by sulth1** [logancyang/obsidian-copilot#395] [Ideas]
Hi! 
It seems that many Obsidian users are using the Zettelkasten method, aka having a bunch of atomic notes. 
Now that Anthropic released cheap models with a context windows of 200 000 tokens, it seems feasible to increase the max sources by a few magnitudes (close to 100? no limit?), or allow it to use all notes in a given folder. (It may of course not be, there may be other factors in play that I'm not aware of.)

As an example, in my case, I have a folder of around 100 atomics notes all on the same specific topic, for a total of approx 110 000 tokens "only". The current maximum of 10 max sources is unfortunately far from enough in an ideal case, and doesn't use the potential of the 200 000 context that Haiku & Sonnet provide. 

Have a great day!

**Comment by logancyang** [comment]
Hi @sulth1, thanks for the suggestion. I understand the need for more source notes, it will be increased. The current setup is mostly for the BETA version.

Just FYI, the increase of number of sources requires more reliable hybrid retrieval and reranking, not just a long context window. I'll first address that and make sure the QA quality doesn't get negatively impacted.

**Reply by ReliablyAwkward** [reply]
Tell me about it! My vault is now 4.5 million words long across about 2,000 notes, all geared towards one project, so to call it vault QA mode but only have it limited to 10 sources is something like, True clickbait, lol....Also, I was using the API key from Cohere for the embedding, which one index cost about 2.3 million tokens, And I figured I could use the embeddings saved local to my machine elsewhere but I can't seem to find them for the life of me, where are they! I already looked in my windows file explorer and searched in beddings and whatnot and nothing, where the fudge is it!

**Reply by sulth1** [reply]
Hey! Any news about it? Now that we have models with 1M token, it would be great. Have a nice day!

**Comment by VladUZH** [comment]
The second part of the proposal, forcing the copilot to chat with a given (list) of folders, is another extremely practical feature. I have a dedicated folder where I keep a list of interviews. And there is another directory where I keep all my notes that I am writing after revising the interviews. Not restricting Copilot to the given folder messes up with desired information extraction. It simply provides me with the content of the notes that I wrote myself.
